### ISSUE TITLE
UNION query return error

### ISSUE THREAD

**Issue by mchades** [StarRocks/starrocks#5711]
### Steps to reproduce the behavior (Required)
1. CREATE TABLE '...'
```
CREATE TABLE lmh.`test` (
  `id` bigint(20),
  `name` varchar(65533),
  `dt` date NULL COMMENT ""
) ENGINE=OLAP
DUPLICATE KEY(`id`)
COMMENT "OLAP"
PARTITION BY RANGE(dt)(
   PARTITION p20220323 VALUES LESS THAN ("2022-03-24"),
   PARTITION p20220324 VALUES LESS THAN ("2022-03-25"),
   PARTITION p20220325 VALUES LESS THAN ("2022-03-26")
)
DISTRIBUTED BY HASH(`id`) BUCKETS 1
PROPERTIES (
  "replication_num" = "1"
);
```
2. INSERT INTO '....'
```
insert into lmh.test VALUES(1, 'test', '2022-03-23'), (2, 'test', '2022-03-24'), (3, 'test', '2022-03-25');
```
3. SELECT ... union ...
```
SELECT count(1) AS b
FROM (
    SELECT max(name) AS a
    FROM lmh.test
  ) t2
WHERE a = '1'
UNION
SELECT 1
FROM lmh.test;
```
### Expected behavior (Required)
return nomally
### Real behavior (Required)
> ERROR 1064 (HY000): low value must be less than or equal to high value or both values have to be NaN, got 0.0 and NaN respectively
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
mysql> select current_version();
+-------------------+
| current_version() |
+-------------------+
| UNKNOWN bae30ec   |
+-------------------+

```

It seems because `ColumnStatistic` does not deal with string max/min

**Comment by mchades** [comment]
It seems branch-2.2 has fixed

**Comment by mchades** [comment]
fixed by #5982
